README: be more fair with all browser vendors

bnjbvr opened the pull request:
Hi there,

The readme as in its current state feels a bit unfair: it makes it feel that only V8 has proper, optimized support for wasm, which is **not** true. Microsoft and [Mozilla](https://bugzilla.mozilla.org/show_bug.cgi?id=1188259) also have VM engines, with different strategies, but all of them including highly optimizing JIT. Apple's Safari support has not been announced in any official statement, although there is some work ongoing in the JSC virtual machine, as shown by this [meta-bug](https://bugs.webkit.org/show_bug.cgi?id=146064). For this reason I have not included JSC in the list, but made it clear that this list was not exhaustive.

Please let me know if you think this needs precisions.

Cheers!

miohtama commented:
The old text indeed would have been a twisted truth as WebAssembly originates from Mozilla and naturally was first to implement.

axic commented:
@bnjbvr can you rework it to be a bit shorter?

There is no malicious intent in that text and I don't think it suggest that it originates from Google. The reason that is listed, because the project at its current state runs on Node.js only.

axic commented:
Here's my version:

```diff
-* Multiple JIT or interpreted VMs exist for WASM. Notably the V8 Javascript engine includes one, which is used in both Node.js and Chrome. Firefox has its own VM implementation.
+* Multiple JIT or interpreted VMs exist for WASM. All major browsers have (or expected to have) support for WASM. Additionally, Node.js implements it too.
```

wanderer commented:
@bnjbvr thanks for the PR! I think it is an important thing to try to communicate here that there are multiple independent implementation of wasm. I like listing the major implementations but maybe drop the "used in"  part. Also it would be nice to have links to the implementations.

bnjbvr commented:
@axic, I think it's slightly better to precise what the VM engines are, because often they're not only used in browsers (e.g. v8 is used in node, spidermonkey in couchdb, etc.) and your change still emphasizes v8 more than the other ones (until node.js gets a JS-engine agnostic implementation).

@wanderer, I've updated the paragraph to remove the "used to", making this slightly shorter. By having links to the implementations, did you mean links to v8 / gecko (firefox's rendering engine since spidermonkey is in there) / chakra?

wanderer commented:
@bnjbvr yep links to v8 / gecko / chakra would be super nice

axic commented:
I think if we're going lengths at listing all implementations, we shouldn't omit these two:
- https://github.com/WebAssembly/wasm-jit-prototype
- https://github.com/WebAssembly/wabt
- (perhaps even WebAssembly/spec?)

And probably we should list them in alphabetical order for simplicity (unless there are better suggestions).

bnjbvr commented:
Here we go! Adding links to non webby implementations makes sense too, but I've split those in a separate list to not overload the first one. I've also rebased all the commits. Can you please have a look, @axic @wanderer ?

wanderer commented:
@bnjbvr thanks looks good to me
